src/ring_theory/polynomial/eisenstein.lean

```diff
@@ -194,6 +195,55 @@ end is_eisenstein_at
 
 end polynomial
 
+section cyclotomic
+
+variables {p : ℕ}
+
+local notation `𝓟` := submodule.span ℤ {p}
+
+open polynomial
+
+lemma cyclotomic_comp_X_add_one_is_eisenstein_at [hp : fact p.prime] :
```

Comment:
I can state this using `geom_sum`, but I feel the statement with `cyclotomic p ` is more natural.